### ISSUE TITLE
Allow for using other field besides 'sub' such as 'uid'

### ISSUE THREAD

**Issue by jessica-mulein** [tymondesigns/jwt-auth#877]
I was hoping to use firebase-compatible tokens for my app, and it seems they need 'sub' to be an email and instead want to use a 'uid' field.

https://firebase.google.com/docs/auth/server/create-custom-tokens

If I change my getJwtIdentifier() function to return the email address firebase wants, I can auth with firebase fine, but jwt-auth package is unhappy when it tries to use sub to get the ID back.

I might make a pull-request as an idea, but suspect you may have a more preferred way to allow this.

**Comment by BrettTrotter** [comment]
Taking a quick stab at this, I first looked at making another function in JWTSubject that was getJwtIdentifierField() but then this could not easily be known during JWTAuth. Then I looked at adding a config item like JWT_IDENTIFIER_CLAIM_FIELD (defaulted to 'sub'), but config is not really easily accessible anywhere in JWT or JWTAuth where we'd need them. I was looking at passing the config data into the manager since the manager seems to be potentially around when we need it.. But these types of changes are going to be very subjective as to how you want to implement and I don't want to mess up your design ideals. I'll keep poking at something and I may submit a PR even if I'm ultimately sure you won't like the direction I took.
